Formula/mono.rb

```diff
@@ -87,9 +93,12 @@ class Mono < Formula
   end
 
   def install
+    # Regenerate configure to fix ARM build: no member named '__r' in 'struct __darwin_arm_thread_state64'
+    system "autoreconf", "--force", "--install", "--verbose" if Hardware::CPU.arm?
```

Comment:
Sadly, although we can build the main `mono`, we aren't able to build `msbuild` resource which may be due to bootstrap. Not sure if there is a way to fix.